pkgs/by-name/ex/ext3grep/package.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchurl
+, e2fsprogs
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ext3grep";
+  version = "0.10.2";
+
+  src = fetchurl {
+    url = "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/ext3grep/ext3grep-${finalAttrs.version}.tar.gz";
+    hash = "sha256-WG8+k50v/XgvbwBrgaPfLcR3xtoD8h7biGDFxPcZjz4=";
+  };
+
+  nativeBuildInputs = [ e2fsprogs ];
+
+  patches = [
+    ./ext3grep-build.patch
+    ./ext3grep-e2fsprogs-1.44.1.patch
+  ];
```

Comment:
these patches look like they might be from the debian distribution?
use fetchpatch to pull them, should be able to access them from
https://salsa.debian.org/pkg-security-team/ext3grep/-/tree/debian/master/debian/patches?ref_type=heads
and can grep nixpkgs for `salsa.debian.org` for some examples.

Reply:
I retrieved those two files from [Arch Linux repository](https://gitlab.archlinux.org/archlinux/packaging/packages/ext3grep) but I cannot find the same patches in that Debian link. Should do apply also the salsa.debian patches?

Reply:
they are the first two patches listed in the debian [link](https://salsa.debian.org/pkg-security-team/ext3grep/-/tree/debian/master/debian/patches?ref_type=heads)
1. https://salsa.debian.org/pkg-security-team/ext3grep/-/blob/debian/master/debian/patches/001_fix-ftbfs-e2fsprogs_1.42-WIP-702.diff?ref_type=heads
2. https://salsa.debian.org/pkg-security-team/ext3grep/-/blob/debian/master/debian/patches/002_remove_i_dir_acl.diff?ref_type=heads

Reply:
> I retrieved those two files from [Arch Linux repository](https://gitlab.archlinux.org/archlinux/packaging/packages/ext3grep) but I cannot find the same patches in that Debian link. Should do apply also the salsa.debian patches?

you can also use fetchpatch to pull them from the arch repo too. there are likely examples in nixpkgs if you grep for the base url.

Reply:
If those patches are "general purpose", then they should be "upstreamed".